Add a tag cloud to the readings index

Chemaclass opened the pull request:
Adds a **Browse by tag** cloud to the readings index.

Re-applied on the current `main` rather than rebased. The 2023 draft counted tags in the browser (a JS loop pushed every tag into an array, injected a `<ul>`, left a `console.log` behind, and styled the chips with `float: left`), and it touched `sass/listing.scss`, `sass/pages/_readings.scss` and `sass/layouts/_taxonomy.scss`, all three of which the quality passes have since rewritten. Replaying the old diff would have fought every one of them.

## What it does

- **Rendered in Tera, no JS.** Taxonomy terms span blog posts, readings and talks, so every term is re-counted against the pages under `/readings/`.
- **Only the two heaviest size tiers show** (15 chips), so the first book stays above the fold. The long tail keeps its home on `/tags/`, linked from the cloud.
- **Chip styles are now shared.** The pill design existed only as `.tags-page__tag`; it moves to `sass/components/_tag-chips.scss` (`.tag-cloud` / `.tag-chip`) and both the all-tags page and the readings bar use it.
- **Dark-mode contrast:** the `--lg` count badge used white on the light accent (about 2.4:1) and now uses the page background.
- New `browse_by_tag` key, EN + ES.

No `!important` needed: `.tag-chip:hover` (0,2,0) already outranks the global prose-underline rule at (0,1,1).

## Verified

```
/readings/       15 chips, label "Browse by tag"
/es/readings/    15 chips, label "Explora por etiqueta"
/readings/page/2 no cloud (page 1 only)
/tags/           174 chips, 0 stale tags-page__tag classes
```

Compiled CSS against `main`: only `listing.css` changes, 12 rules removed (the old `tags-page__tag*` set) and 13 added (`.tag-chip*`, the readings bar, the dark-mode override). `page.css`, `post.css` and `index.css` are rule-for-rule identical. `./build.sh` exits 0.

## One mistake worth recording

My first pass inserted the `.readings-tags` styles by targeting an empty `.readings-index {}` selector, which the comment pass had already deleted, and I did not assert the replace. It silently did nothing, and the bar shipped with an unstyled label, left aligned. The screenshot caught it, not the build, because missing CSS breaks nothing. Fixed in a follow-up commit.

CosmeValera commented:
It looks like a good idea for me:
![image](https://github.com/Chemaclass/chemaclass.com/assets/80126839/bac6937f-c04e-4da2-91ec-3e779f0d3d67)

Visually, maybe it would be a bettter idea to have them more like actual tags, e.g: [PrimeReact tags](https://primereact.org/tag/):
![image](https://github.com/Chemaclass/chemaclass.com/assets/80126839/60d8f874-1663-4e21-9801-916ecd10da47)
![image](https://github.com/Chemaclass/chemaclass.com/assets/80126839/42f5ec30-1954-4eb7-a3fe-e2165a5e32dd)

Also, since they take up a lot of space. It could be a good idea to put them inside a container, that could be toggleable. Like the [fieldset](https://primereact.org/fieldset/) component of Prime React.
![image](https://github.com/Chemaclass/chemaclass.com/assets/80126839/8e7c7681-2609-4380-9115-7a93fa45c1d6)
*Instead of "header", it could be "tags".

And another feature idea, is that you could even use the tags to filter what books to show, this way implementing a simple js, you could filter to show only the books of type "leadership", or of type "people" just by clicking the tag.

Chemaclass committed:
feat(readings): add a tag cloud to the readings index

Re-applied on the current main rather than rebased: the 2023 draft touched
sass/listing.scss, sass/pages/_readings.scss and sass/layouts/_taxonomy.scss, all
of which the quality passes rewrote, and it built the list in the browser.

Rendered in Tera. Taxonomy terms span blog posts, readings and talks, so each term
is re-counted against the pages under /readings/, and only the two heaviest size
tiers show so the first book stays above the fold. The long tail keeps its home on
the existing /tags/ page, linked from the cloud.

The chip styling already existed as .tags-page__tag on the all-tags cloud, so it
moves to components/tag-chips (.tag-cloud / .tag-chip) and both pages use it. The
--lg count badge uses the page background in dark mode, where white text on the
light accent sat near 2.4:1.

Chemaclass committed:
fix(readings): add the tag bar styles that a silent replace dropped

The insert targeted an empty .readings-index {} selector that the comment pass had
already removed, and the replace was not asserted, so the styles never landed: the
label rendered unstyled and the bar was left aligned instead of centred.
